From: Petr Tesarik <petr@tesarici.cz>
Subject: Provide a fallback definition of EM_RISCV
Upstream: never, compatibility with SLE12

Fix build if <elf.h> does not define EM_RISCV (yet).

Signed-off-by: Petr Tesarik <petr@tesarici.cz>

---
 diskdump.h |    4 ++++
 netdump.h  |    4 ++++
 ramdump.c  |    4 ++++
 symbols.c  |    4 ++++
 4 files changed, 16 insertions(+)

--- a/diskdump.h
+++ b/diskdump.h
@@ -19,6 +19,10 @@
 
 #include <elf.h>
 
+#ifndef EM_RISCV
+#define EM_RISCV 243
+#endif
+
 #define divideup(x, y)	(((x) + ((y) - 1)) / (y))
 #define round(x, y)	(((x) / (y)) * (y))
 
--- a/netdump.h
+++ b/netdump.h
@@ -19,6 +19,10 @@
 #include <elf.h>
 #include "vmcore.h"
 
+#ifndef EM_RISCV
+#define EM_RISCV 243
+#endif
+
 #define MIN_NETDUMP_ELF32_HEADER_SIZE \
         sizeof(Elf32_Ehdr)+sizeof(Elf32_Phdr)+sizeof(Elf32_Phdr)
 #define MIN_NETDUMP_ELF64_HEADER_SIZE \
--- a/ramdump.c
+++ b/ramdump.c
@@ -22,6 +22,10 @@
 #include "defs.h"
 #include <elf.h>
 
+#ifndef EM_RISCV
+#define EM_RISCV 243
+#endif
+
 struct ramdump_def {
 	char *path;
 	int rfd;
--- a/symbols.c
+++ b/symbols.c
@@ -23,6 +23,10 @@
 #endif
 #include "bfd.h"
 
+#ifndef EM_RISCV
+#define EM_RISCV 243
+#endif
+
 static void store_symbols(bfd *, int, void *, long, unsigned int);
 static void store_sysmap_symbols(void);
 static ulong relocate(ulong, char *, int);
